### DISCUSSION TITLE
Use vcpkg from Windows Docker container with vs_buildtools

### DISCUSSION THREAD

**Discussion by dstaley** [microsoft/vcpkg#25353] [Q&A]
I've made several attempts at building a Docker container that installs the Visual Studio Build Tools with `vs_buildtools.exe` and then tries to install libraries with `vcpkg`.

My most recent attempt used the following installation command:
```
buildtools.exe --quiet --wait --norestart --nocache modify 
  --installPath "%ProgramFiles(x86)%\Microsoft Visual Studio\2022\BuildTools"
  --add Microsoft.VisualStudio.Workload.UniversalBuildTools
  --add Microsoft.VisualStudio.Workload.VCTools
  --includeRecommended
  --includeOptional
```

When attempting to install a library using the `arm64-windows` triplet, I'd get the following error:
```
Detecting compiler hash for triplet "x64-windows"...
Detecting compiler hash for triplet "arm64-windows"...
error: in triplet arm64-windows: Unable to find a valid toolchain for requested target architecture arm64.
The selected Visual Studio instance is at: C:\Program Files (x86)\Microsoft Visual Studio 14.0\
The available toolchain combinations are: x86, x64, x86_amd64, x86_arm, amd64_x86, amd64_arm
```

When looking at `C:\Program Files (x86)`, I'd actually have two folders: `Microsoft Visual Studio` and `Microsoft Visual Studio 14.0`, and it looks like `vcpkg` was defaulting to the latter which I don't believe uses the `arm64` architecture (instead using `amd64_arm`).

If anyone has managed to install the Visual Studio Build Tools in a Windows Docker container in a way that satisfies `vcpkg` I'd love to hear what your installation command was, and if there's any additional setup you did to get things working.

**Comment by thxkiwi** [comment]
Not exactly in the same boat (I'm just wanting to use x64-windows and x32-windows from within a container) but I've had considerable issues with trying to get vcpkg to use a particular install of the Windows toolchain, and the closest I came was https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_visual_studio_path but still could not get this to work with either a container and vsbuildtools or, in my case, the EWDK for building drivers.

**Reply by thxkiwi** [reply]
@dstaley what's your vcpkg host architecture? Is there actually support for build arm64 from your host architecture?
